micromanager_gui/multid_widget.py

```diff
@@ -249,9 +272,25 @@ def _on_run_clicked(self):
             print("Select at least one channel.")
             return
 
-        experiment = MDASequence(**self._get_state_dict())
-        self._mmc.run_mda(experiment)  # run the MDA experiment asynchronously
-        return
+        if self.save_groupBox.isChecked(): 
+
+            if self.fname_lineEdit.text() == '' or \
+                (self.dir_lineEdit.text() == '' or \
+                    not Path.is_dir(Path(self.dir_lineEdit.text()))
+                    ):
+                        raise ValueError ('select a filename and a valid directory.')
+                        print('select a filename and a valid directory.')
+
+            else:
+                experiment = MDASequence(**self._get_state_dict())
+                self._mmc.run_mda(experiment)  # run the MDA experiment asynchronously
+                return
+
+        else:
+            experiment = MDASequence(**self._get_state_dict())
+            self._mmc.run_mda(experiment)  # run the MDA experiment asynchronously
+            return
+
```

Comment:
refactor this to remove the code duplication... also, you don't need to use if/else if you're going to raise an error.  so this can just be something like
```python
if self.save_groupBox.isChecked() and some_other_condition_is_not_met:
    raise ValueError ('select a filename and a valid directory.')

experiment = MDASequence(**self._get_state_dict())
self._mmc.run_mda(experiment)  # run the MDA experiment asynchronously
return
```

Reply:
Thanks!